### ISSUE TITLE
Adjusting the Sonarqube Quality Gate

### ISSUE THREAD

**Issue by javahippie** [operaton/operaton#213]
**Is your feature request related to a problem? Please describe.**
We are often changing existing code, which does not have a lot of coverage. While the overall code coverage is good, many of our merge requests are being displayed as "failed", because we edit code which would not neccessarily benefit from code coverage, e.g. data classes.

My proposal is to check the code coverage not for new code, but globally, so our build history does not look like it is failing all the time, raising impressions of sloppy work.

**Comment by javahippie** [comment]
@kthoms @a-m-zill @hauptmedia Opinions on this?

**Comment by hauptmedia** [comment]
I’ve noticed that the pipeline is currently always marked as “red,” and I believe this creates a negative first impression of the project. It can be misleading, especially considering that over 30,000 tests are being executed successfully. This doesn’t seem to accurately reflect the project’s overall health.

I suggest reviewing the pipeline settings to better align with the project’s success criteria.

**Comment by kthoms** [comment]
At this moment it is quite likely that we fail the criteria. For example, I just changed a single line of source in source/main amongst a ton of changes in tests, and Sonar made this fail the quality gate as the single line was not covered by a test (=> 0% coverage). I have ignored this fact and just merged.

Also there could be that one change is degrading the code quality, but I'm cleaning this up with later changes.

For now I would be OK to lower the quality gate, but once we are done with the cleanups we should enforce that again. Definetely before the 1.0.0 Final version.
